lib/pf/api.pm

```diff
@@ -61,22 +61,53 @@ use pf::dhcp::processor();
 use pf::util::dhcpv6();
 
 sub event_add : Public {
-    my ($class, $date, $srcip, $type, $id) = @_;
+    my ($class, %postdata) = @_;
     my $logger = pf::log::get_logger();
-    $logger->info("violation: $id - IP $srcip");
-
-    # fetch IP associated to MAC
-    my $srcmac = pf::iplog::ip2mac($srcip);
-    if ($srcmac) {
-
-        # trigger a violation
-        pf::violation::violation_trigger( { 'mac' => $srcmac, 'tid' => $id, 'type' => $type } );
 
+    my ($type, $id) = each %{$postdata{'events'}};
+    $logger->info("violation: $id - IP $postdata{'srcip'}") if (defined($postdata{'srcip'}));
+    if (defined ($pf::config::Config{'trapping'}{'range'}) && $pf::config::Config{'trapping'}{'range'} ne '') {
+        foreach my $network (split(',',$pf::config::Config{'trapping'}{'range'})) {
+            my $net_addr = NetAddr::IP->new($network);
```

Comment:
These objects should be built in a pfconfig resource

Otherwise we have the cost of creating the object of each network every time we parse an alert